src/components/Map/Layers/SymbolLayer.jsx

```diff
@@ -0,0 +1,120 @@
+import { useEffect, useState } from 'react';
+import OLVectorLayer from 'ol/layer/Vector';
+import {
+  Style,
+  Icon,
+  Fill,
+  Text,
+} from 'ol/style';
+import Feature from 'ol/Feature';
+import Point from 'ol/geom/Point';
+import { fromLonLat } from 'ol/proj';
+// import { rgba } from '../../MapVizzard/MapOptions/ColorPicker';
+import { vector } from '../Source';
+import capital from '../assets/map-icons/capital.svg';
+import city from '../assets/map-icons/city.svg';
+import settlement from '../assets/map-icons/settlement.svg';
+import marker from '../assets/map-icons/marker.svg';
+import airport from '../assets/map-icons/airport.svg';
+import idpRefugeeCamp from '../assets/map-icons/idp-refugee-camp.svg';
+
+function SymbolLayer({
+  map,
+  source,
+  data,
+  symbol,
+  zIndex = 1,
+  opacity = 1,
+  showLabels = false,
+  labelColumn = '',
+  scale = 1,
+  textScale = 1,
+}) {
+  const [symbolLayer, setSymbolLayer] = useState(false);
+
+  const symbolIcons = {
+    capital,
+    city,
+    settlement,
+    'idp-refugee-camp': idpRefugeeCamp,
+    airport,
+    marker,
+  };
+
+  useEffect(() => {
+    if (!map) return undefined;
+
+    let xOffset = 1.7;
+
+    if (symbol === 'capital') xOffset = 0.7;
+
+    const features = data.map((item) => {
+      const feature = new Feature({
+        geometry: new Point(fromLonLat([item.lon, item.lat])),
+      });
+
+      const iconStyle = [
+        new Style({
+          image: new Icon({
+            anchor: [0.5, 0.5],
+            anchorXUnits: 'fraction',
+            anchorYUnits: 'fraction',
+            scale,
+            src: symbolIcons[symbol],
+          }),
+        }),
+      ];
+
+      if (showLabels === true) {
+        const label = (item[labelColumn]) || '';
```

Comment:
We should use `??` instead of `||` to pass default values.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing

`0 || 1` will be `1` but the expected outcome is `0`
`0 ?? 1` will be `0`